app/services/invoices/payments/cashfree_service.rb

```diff
@@ -0,0 +1,202 @@
+
+        result.payment = payment
+        result
+      end
+
+      def update_payment_status(provider_payment_id:, status:, metadata: {})
+        payment = if metadata[:payment_type] == 'one-time'
+          create_payment(provider_payment_id:)
+        else
+          Payment.find_by(provider_payment_id:)
+        end
+        return result.not_found_failure!(resource: 'cashfree_payment') unless payment
+
+        result.payment = payment
+        result.invoice = payment.payable
+        return result if payment.payable.payment_succeeded?
+
+        invoice_payment_status = invoice_payment_status(status)
+
+        payment.update!(status: invoice_payment_status)
+        update_invoice_payment_status(payment_status: invoice_payment_status)
+
+        result
+      rescue BaseService::FailedResult => e
+        result.fail_with_error!(e)
+      end
+
+      def generate_payment_url
+        return result unless should_process_payment?
+
+        res = create_post_request(payment_url_params)
+
+        result.payment_url = JSON.parse(res.body)["link_url"]
+
+        result
+      rescue LagoHttpClient::HttpError => e
+        deliver_error_webhook(e)
+        result.service_failure!(code: e.error_code, message: e.error_body)
+      end
+
+      private
+
+      attr_accessor :invoice
+
+      delegate :organization, :customer, to: :invoice
+
+      def create_payment(provider_payment_id:)
+        @invoice = Invoice.find_by(id: provider_payment_id)
+
+        increment_payment_attempts
+
+        Payment.new(
+          payable: invoice,
+          payment_provider_id: cashfree_payment_provider.id,
+          payment_provider_customer_id: customer.cashfree_customer.id,
+          amount_cents: invoice.total_amount_cents,
+          amount_currency: invoice.currency.upcase,
+          provider_payment_id:
+        )
+      end
+
+      def should_process_payment?
+        return false if invoice.payment_succeeded? || invoice.voided?
+        return false if cashfree_payment_provider.blank?
+
+        customer&.cashfree_customer&.id
+      end
+
+      def client
+        @client ||= LagoHttpClient::Client.new(::PaymentProviders::CashfreeProvider::BASE_URL)
+      end
+
+      def create_post_request(body)
+        client.post_with_response(body, {
+          "accept" => 'application/json',
+          "content-type" => 'application/json',
+          "x-client-id" => cashfree_payment_provider.client_id,
+          "x-client-secret" => cashfree_payment_provider.client_secret,
+          "x-api-version" => ::PaymentProviders::CashfreeProvider::API_VERSION
+        })
+      end
+
+      def success_redirect_url
+        cashfree_payment_provider.success_redirect_url.presence || ::PaymentProviders::CashfreeProvider::SUCCESS_REDIRECT_URL
+      end
+
+      def cashfree_payment_provider
+        @cashfree_payment_provider ||= payment_provider(customer)
+      end
+
+      def payment_url_params
+        # TODO Add settings_accessors if required
```

Comment:
Does it need to be addressed?

Reply:
No, this is not required as all the payment handling should be managed through Lago. The Payment URLs are short-lived, so the settings I considered don't make sense.

Reply:
Ok great! So I guess you can get rid of the comment